packages/components/src/tooltip/index.js

```diff
@@ -24,60 +25,69 @@ import Shortcut from '../shortcut';
  *
  * @type {number}
  */
-const TOOLTIP_DELAY = 700;
-
-class Tooltip extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.delayedSetIsOver = debounce(
-			( isOver ) => this.setState( { isOver } ),
-			TOOLTIP_DELAY
-		);
-
-		/**
-		 * Prebound `isInMouseDown` handler, created as a constant reference to
-		 * assure ability to remove in component unmount.
-		 *
-		 * @type {Function}
-		 */
-		this.cancelIsMouseDown = this.createSetIsMouseDown( false );
-
-		/**
-		 * Whether a the mouse is currently pressed, used in determining whether
-		 * to handle a focus event as displaying the tooltip immediately.
-		 *
-		 * @type {boolean}
-		 */
-		this.isInMouseDown = false;
-
-		this.state = {
-			isOver: false,
-		};
-	}
+export const TOOLTIP_DELAY = 700;
 
-	componentWillUnmount() {
-		this.delayedSetIsOver.cancel();
+const emitToChild = ( children, eventName, event ) => {
+	if ( Children.count( children ) !== 1 ) {
+		return;
+	}
 
-		document.removeEventListener( 'mouseup', this.cancelIsMouseDown );
+	const child = Children.only( children );
+	if ( typeof child.props[ eventName ] === 'function' ) {
+		child.props[ eventName ]( event );
 	}
+};
 
-	emitToChild( eventName, event ) {
-		const { children } = this.props;
-		if ( Children.count( children ) !== 1 ) {
-			return;
-		}
+function Tooltip( { children, position, text, shortcut } ) {
+	/**
+	 * Whether a mouse is currently pressed, used in determining whether
+	 * to handle a focus event as displaying the tooltip immediately.
+	 *
+	 * @type {boolean}
+	 */
+	const [ isMouseDown, setIsMouseDown ] = useState( false );
+	const [ isOver, setIsOver ] = useState( false );
+	const delayedSetIsOver = debounce( setIsOver, TOOLTIP_DELAY );
```

Comment:
I think this might create a new debounced function on every render. There is a `useDebounce` hook in the `@wordpress/compose` package that might help, that seems to memoize the call to debounce and also automatically calls `cancel` on unmount.

Reply:
it does create debounce in every rerender this is why I have used

```
	const clearOnUnmount = () => {
		delayedSetIsOver.cancel();
		document.removeEventListener( 'mouseup', cancelIsMouseDown );
	};
useEffect( () => clearOnUnmount, [] );
```

if useDebounce is the preferred way to handle such situations I will use it here